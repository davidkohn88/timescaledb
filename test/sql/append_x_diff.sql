-- Copyright (c) 2016-2018  Timescale, Inc. All Rights Reserved.
--
-- This file is licensed under the Apache License,
-- see LICENSE-APACHE at the top level directory.

--make sure diff only has explain output not result output
\! diff ${TEST_OUTPUT_DIR}/results/append_unoptimized.out ${TEST_OUTPUT_DIR}/results/append.out
