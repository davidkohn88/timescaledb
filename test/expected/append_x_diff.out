-- Copyright (c) 2016-2018  Timescale, Inc. All Rights Reserved.
--
-- This file is licensed under the Apache License,
-- see LICENSE-APACHE at the top level directory.
--make sure diff only has explain output not result output
\! diff ${TEST_OUTPUT_DIR}/results/append_unoptimized.out ${TEST_OUTPUT_DIR}/results/append.out
5c5
< SET timescaledb.disable_optimizations = ON;
---
> SET timescaledb.disable_optimizations = OFF;
58,69c58,64
<                                     QUERY PLAN                                    
< ----------------------------------------------------------------------------------
<  Append
<    ->  Seq Scan on append_test
<          Filter: ("time" > (now_s() + '@ 1 mon'::interval))
<    ->  Index Scan using _hyper_1_1_chunk_append_test_time_idx on _hyper_1_1_chunk
<          Index Cond: ("time" > (now_s() + '@ 1 mon'::interval))
<    ->  Index Scan using _hyper_1_2_chunk_append_test_time_idx on _hyper_1_2_chunk
<          Index Cond: ("time" > (now_s() + '@ 1 mon'::interval))
<    ->  Index Scan using _hyper_1_3_chunk_append_test_time_idx on _hyper_1_3_chunk
<          Index Cond: ("time" > (now_s() + '@ 1 mon'::interval))
< (9 rows)
---
> psql:include/append.sql:47: NOTICE:  Stable function now_s() called!
>              QUERY PLAN              
> -------------------------------------
>  Custom Scan (ConstraintAwareAppend)
>    Hypertable: append_test
>    Chunks left after exclusion: 0
> (3 rows)
77,78d71
< psql:include/append.sql:48: NOTICE:  Stable function now_s() called!
< psql:include/append.sql:48: NOTICE:  Stable function now_s() called!
91,92c84,86
<                                        QUERY PLAN                                       
< ----------------------------------------------------------------------------------------
---
> psql:include/append.sql:53: NOTICE:  Stable function now_s() called!
>                 QUERY PLAN                 
> -------------------------------------------
94,104c88,91
<    ->  Merge Append
<          Sort Key: append_test."time" DESC
<          ->  Index Scan using append_test_time_idx on append_test
<                Index Cond: ("time" > (now_s() + '@ 1 mon'::interval))
<          ->  Index Scan using _hyper_1_1_chunk_append_test_time_idx on _hyper_1_1_chunk
<                Index Cond: ("time" > (now_s() + '@ 1 mon'::interval))
<          ->  Index Scan using _hyper_1_2_chunk_append_test_time_idx on _hyper_1_2_chunk
<                Index Cond: ("time" > (now_s() + '@ 1 mon'::interval))
<          ->  Index Scan using _hyper_1_3_chunk_append_test_time_idx on _hyper_1_3_chunk
<                Index Cond: ("time" > (now_s() + '@ 1 mon'::interval))
< (11 rows)
---
>    ->  Custom Scan (ConstraintAwareAppend)
>          Hypertable: append_test
>          Chunks left after exclusion: 0
> (4 rows)
116,127c103,112
<                                     QUERY PLAN                                    
< ----------------------------------------------------------------------------------
<  Append
<    ->  Seq Scan on append_test
<          Filter: ("time" > (now_s() - '@ 2 mons'::interval))
<    ->  Index Scan using _hyper_1_1_chunk_append_test_time_idx on _hyper_1_1_chunk
<          Index Cond: ("time" > (now_s() - '@ 2 mons'::interval))
<    ->  Index Scan using _hyper_1_2_chunk_append_test_time_idx on _hyper_1_2_chunk
<          Index Cond: ("time" > (now_s() - '@ 2 mons'::interval))
<    ->  Index Scan using _hyper_1_3_chunk_append_test_time_idx on _hyper_1_3_chunk
<          Index Cond: ("time" > (now_s() - '@ 2 mons'::interval))
< (9 rows)
---
> psql:include/append.sql:60: NOTICE:  Stable function now_s() called!
>                                        QUERY PLAN                                       
> ----------------------------------------------------------------------------------------
>  Custom Scan (ConstraintAwareAppend)
>    Hypertable: append_test
>    Chunks left after exclusion: 1
>    ->  Append
>          ->  Index Scan using _hyper_1_3_chunk_append_test_time_idx on _hyper_1_3_chunk
>                Index Cond: ("time" > (now_s() - '@ 2 mons'::interval))
> (6 rows)
137d121
< psql:include/append.sql:63: NOTICE:  Stable function now_s() called!
152,153c136,138
<                                            QUERY PLAN                                            
< -------------------------------------------------------------------------------------------------
---
> psql:include/append.sql:69: NOTICE:  Stable function now_s() called!
>                                               QUERY PLAN                                               
> -------------------------------------------------------------------------------------------------------
155,165c140,147
<    ->  Merge Append
<          Sort Key: append_test."time"
<          ->  Index Scan Backward using append_test_time_idx on append_test
<                Index Cond: ("time" > (now_s() - '@ 2 mons'::interval))
<          ->  Index Scan Backward using _hyper_1_1_chunk_append_test_time_idx on _hyper_1_1_chunk
<                Index Cond: ("time" > (now_s() - '@ 2 mons'::interval))
<          ->  Index Scan Backward using _hyper_1_2_chunk_append_test_time_idx on _hyper_1_2_chunk
<                Index Cond: ("time" > (now_s() - '@ 2 mons'::interval))
<          ->  Index Scan Backward using _hyper_1_3_chunk_append_test_time_idx on _hyper_1_3_chunk
<                Index Cond: ("time" > (now_s() - '@ 2 mons'::interval))
< (11 rows)
---
>    ->  Custom Scan (ConstraintAwareAppend)
>          Hypertable: append_test
>          Chunks left after exclusion: 1
>          ->  Merge Append
>                Sort Key: _hyper_1_3_chunk."time"
>                ->  Index Scan Backward using _hyper_1_3_chunk_append_test_time_idx on _hyper_1_3_chunk
>                      Index Cond: ("time" > (now_s() - '@ 2 mons'::interval))
> (8 rows)
176,177d157
< psql:include/append.sql:73: NOTICE:  Stable function now_s() called!
< psql:include/append.sql:73: NOTICE:  Stable function now_s() called!
190,191c170,171
<                                                           QUERY PLAN                                                          
< ------------------------------------------------------------------------------------------------------------------------------
---
>                                                              QUERY PLAN                                                             
> ------------------------------------------------------------------------------------------------------------------------------------
194,209c174,182
<    ->  Append
<          ->  Seq Scan on append_test
<                Filter: ("time" > ('Tue Aug 22 10:00:00 2017 PDT'::timestamp with time zone - '@ 2 mons'::interval))
<          ->  Bitmap Heap Scan on _hyper_1_1_chunk
<                Recheck Cond: ("time" > ('Tue Aug 22 10:00:00 2017 PDT'::timestamp with time zone - '@ 2 mons'::interval))
<                ->  Bitmap Index Scan on _hyper_1_1_chunk_append_test_time_idx
<                      Index Cond: ("time" > ('Tue Aug 22 10:00:00 2017 PDT'::timestamp with time zone - '@ 2 mons'::interval))
<          ->  Bitmap Heap Scan on _hyper_1_2_chunk
<                Recheck Cond: ("time" > ('Tue Aug 22 10:00:00 2017 PDT'::timestamp with time zone - '@ 2 mons'::interval))
<                ->  Bitmap Index Scan on _hyper_1_2_chunk_append_test_time_idx
<                      Index Cond: ("time" > ('Tue Aug 22 10:00:00 2017 PDT'::timestamp with time zone - '@ 2 mons'::interval))
<          ->  Bitmap Heap Scan on _hyper_1_3_chunk
<                Recheck Cond: ("time" > ('Tue Aug 22 10:00:00 2017 PDT'::timestamp with time zone - '@ 2 mons'::interval))
<                ->  Bitmap Index Scan on _hyper_1_3_chunk_append_test_time_idx
<                      Index Cond: ("time" > ('Tue Aug 22 10:00:00 2017 PDT'::timestamp with time zone - '@ 2 mons'::interval))
< (17 rows)
---
>    ->  Custom Scan (ConstraintAwareAppend)
>          Hypertable: append_test
>          Chunks left after exclusion: 1
>          ->  Append
>                ->  Bitmap Heap Scan on _hyper_1_3_chunk
>                      Recheck Cond: ("time" > ('Tue Aug 22 10:00:00 2017 PDT'::timestamp with time zone - '@ 2 mons'::interval))
>                      ->  Bitmap Index Scan on _hyper_1_3_chunk_append_test_time_idx
>                            Index Cond: ("time" > ('Tue Aug 22 10:00:00 2017 PDT'::timestamp with time zone - '@ 2 mons'::interval))
> (10 rows)
218,219c191,192
<                             QUERY PLAN                             
< -------------------------------------------------------------------
---
>                                QUERY PLAN                                
> -------------------------------------------------------------------------
222,231c195,205
<    ->  Append
<          ->  Seq Scan on append_test
<                Filter: ("time" > (now_v() - '@ 2 mons'::interval))
<          ->  Seq Scan on _hyper_1_1_chunk
<                Filter: ("time" > (now_v() - '@ 2 mons'::interval))
<          ->  Seq Scan on _hyper_1_2_chunk
<                Filter: ("time" > (now_v() - '@ 2 mons'::interval))
<          ->  Seq Scan on _hyper_1_3_chunk
<                Filter: ("time" > (now_v() - '@ 2 mons'::interval))
< (11 rows)
---
>    ->  Custom Scan (ConstraintAwareAppend)
>          Hypertable: append_test
>          Chunks left after exclusion: 3
>          ->  Append
>                ->  Seq Scan on _hyper_1_1_chunk
>                      Filter: ("time" > (now_v() - '@ 2 mons'::interval))
>                ->  Seq Scan on _hyper_1_2_chunk
>                      Filter: ("time" > (now_v() - '@ 2 mons'::interval))
>                ->  Seq Scan on _hyper_1_3_chunk
>                      Filter: ("time" > (now_v() - '@ 2 mons'::interval))
> (12 rows)
257,258d230
< psql:include/append.sql:99: NOTICE:  Stable function now_s() called!
< psql:include/append.sql:99: NOTICE:  Stable function now_s() called!
266,278c238,248
<                                         QUERY PLAN                                         
< -------------------------------------------------------------------------------------------
<  Merge Append
<    Sort Key: append_test."time"
<    ->  Index Scan Backward using append_test_time_idx on append_test
<          Index Cond: ("time" > (now_s() - '@ 2 mons'::interval))
<    ->  Index Scan Backward using _hyper_1_1_chunk_append_test_time_idx on _hyper_1_1_chunk
<          Index Cond: ("time" > (now_s() - '@ 2 mons'::interval))
<    ->  Index Scan Backward using _hyper_1_2_chunk_append_test_time_idx on _hyper_1_2_chunk
<          Index Cond: ("time" > (now_s() - '@ 2 mons'::interval))
<    ->  Index Scan Backward using _hyper_1_3_chunk_append_test_time_idx on _hyper_1_3_chunk
<          Index Cond: ("time" > (now_s() - '@ 2 mons'::interval))
< (10 rows)
---
> psql:include/append.sql:102: NOTICE:  Stable function now_s() called!
>                                            QUERY PLAN                                            
> -------------------------------------------------------------------------------------------------
>  Custom Scan (ConstraintAwareAppend)
>    Hypertable: append_test
>    Chunks left after exclusion: 1
>    ->  Merge Append
>          Sort Key: _hyper_1_3_chunk."time"
>          ->  Index Scan Backward using _hyper_1_3_chunk_append_test_time_idx on _hyper_1_3_chunk
>                Index Cond: ("time" > (now_s() - '@ 2 mons'::interval))
> (7 rows)
306a277
> psql:include/append.sql:115: NOTICE:  Stable function now_s() called!
313c284,286
<          ->  Result
---
>          ->  Custom Scan (ConstraintAwareAppend)
>                Hypertable: append_test
>                Chunks left after exclusion: 2
315,318d287
<                      ->  Seq Scan on append_test
<                            Filter: ("time" > (now_s() - '@ 4 mons'::interval))
<                      ->  Index Scan using _hyper_1_1_chunk_append_test_time_idx on _hyper_1_1_chunk
<                            Index Cond: ("time" > (now_s() - '@ 4 mons'::interval))
323c292
< (14 rows)
---
> (12 rows)
335,336c304,305
<                                                                                                              QUERY PLAN                                                                                                              
< -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
---
>                               QUERY PLAN                               
> -----------------------------------------------------------------------
341,345c310,313
<          ->  Result
<                ->  Append
<                      ->  Seq Scan on append_test
<                            Filter: (("time" < 'Tue Mar 22 00:00:00 2016 PDT'::timestamp with time zone) AND (date_part('dow'::text, "time") >= '1'::double precision) AND (date_part('dow'::text, "time") <= '5'::double precision))
< (8 rows)
---
>          ->  Custom Scan (ConstraintAwareAppend)
>                Hypertable: append_test
>                Chunks left after exclusion: 0
> (7 rows)
390,391c358,360
<                                               QUERY PLAN                                               
< -------------------------------------------------------------------------------------------------------
---
> psql:include/append.sql:154: NOTICE:  Stable function now_s() called!
>                                                   QUERY PLAN                                                   
> ---------------------------------------------------------------------------------------------------------------
397c366,368
<            ->  Result
---
>            ->  Custom Scan (ConstraintAwareAppend)
>                  Hypertable: append_test
>                  Chunks left after exclusion: 3
399,401c370
<                        ->  Seq Scan on append_test
<                              Filter: ((colorid > 0) AND ("time" > (now_s() - '@ 400 days'::interval)))
<                        ->  Index Scan using _hyper_1_1_chunk_append_test_time_idx on _hyper_1_1_chunk
---
>                        ->  Index Scan Backward using _hyper_1_1_chunk_append_test_time_idx on _hyper_1_1_chunk
404c373
<                        ->  Index Scan using _hyper_1_2_chunk_append_test_time_idx on _hyper_1_2_chunk
---
>                        ->  Index Scan Backward using _hyper_1_2_chunk_append_test_time_idx on _hyper_1_2_chunk
407c376
<                        ->  Index Scan using _hyper_1_3_chunk_append_test_time_idx on _hyper_1_3_chunk
---
>                        ->  Index Scan Backward using _hyper_1_3_chunk_append_test_time_idx on _hyper_1_3_chunk
442a412
> psql:include/append.sql:171: NOTICE:  Stable function now_s() called!
482,483c452,455
<                                          QUERY PLAN                                         
< --------------------------------------------------------------------------------------------
---
> psql:include/append.sql:192: NOTICE:  Stable function now_s() called!
> psql:include/append.sql:192: NOTICE:  Stable function now_s() called!
>                                             QUERY PLAN                                            
> --------------------------------------------------------------------------------------------------
486,504c458,474
<    ->  Append
<          ->  Seq Scan on append_test a
<                Filter: ("time" > (now_s() - '@ 3 hours'::interval))
<          ->  Index Scan using _hyper_1_1_chunk_append_test_time_idx on _hyper_1_1_chunk a_1
<                Index Cond: ("time" > (now_s() - '@ 3 hours'::interval))
<          ->  Index Scan using _hyper_1_2_chunk_append_test_time_idx on _hyper_1_2_chunk a_2
<                Index Cond: ("time" > (now_s() - '@ 3 hours'::interval))
<          ->  Index Scan using _hyper_1_3_chunk_append_test_time_idx on _hyper_1_3_chunk a_3
<                Index Cond: ("time" > (now_s() - '@ 3 hours'::interval))
<    ->  Append
<          ->  Seq Scan on join_test j
<                Filter: ("time" > (now_s() - '@ 3 hours'::interval))
<          ->  Index Scan using _hyper_2_4_chunk_join_test_time_idx on _hyper_2_4_chunk j_1
<                Index Cond: ("time" > (now_s() - '@ 3 hours'::interval))
<          ->  Index Scan using _hyper_2_5_chunk_join_test_time_idx on _hyper_2_5_chunk j_2
<                Index Cond: ("time" > (now_s() - '@ 3 hours'::interval))
<          ->  Index Scan using _hyper_2_6_chunk_join_test_time_idx on _hyper_2_6_chunk j_3
<                Index Cond: ("time" > (now_s() - '@ 3 hours'::interval))
< (20 rows)
---
>    ->  Custom Scan (ConstraintAwareAppend)
>          Hypertable: append_test
>          Chunks left after exclusion: 1
>          ->  Append
>                ->  Index Scan using _hyper_1_3_chunk_append_test_time_idx on _hyper_1_3_chunk a_1
>                      Index Cond: ("time" > (now_s() - '@ 3 hours'::interval))
>    ->  Custom Scan (ConstraintAwareAppend)
>          Hypertable: join_test
>          Chunks left after exclusion: 3
>          ->  Append
>                ->  Index Scan using _hyper_2_4_chunk_join_test_time_idx on _hyper_2_4_chunk j_1
>                      Index Cond: ("time" > (now_s() - '@ 3 hours'::interval))
>                ->  Index Scan using _hyper_2_5_chunk_join_test_time_idx on _hyper_2_5_chunk j_2
>                      Index Cond: ("time" > (now_s() - '@ 3 hours'::interval))
>                ->  Index Scan using _hyper_2_6_chunk_join_test_time_idx on _hyper_2_6_chunk j_3
>                      Index Cond: ("time" > (now_s() - '@ 3 hours'::interval))
> (18 rows)
